### DISCUSSION TITLE
Add events in the past or in the future

### DISCUSSION THREAD

**Discussion by Soren-Knudsen** [JasperFx/marten#1494] [Q&A] [answered]
It is posslbe via MartenDB to add events either in the past or in the future?

**Scenario A**: A person "Peter" will change name to "John" at `t_future`.
**Scenario B**: A shop owner finds out that that the expiration date on a product is wrong and wants to correct the mistake back in time from `t_past`.

`t_past` = 01/06/1960
`t_now` = 01/06/2020
`t_future` = 01/06/2125

**Comment by jokokko** [comment, marked as answer]
Not directly, though Copy & Transform might help you out here https://martendb.io/documentation/scenarios/copyandtransformstream/ (though currently it doesn't address event metadata, namely if one wanted to preserve the timestamp field value).

**Comment by Soren-Knudsen** [comment]
Thanks for your quick reply. That could work! Is it possible to chain the streams? When you for example load an aggregate ?

**Comment by jokokko** [comment]
You'd be looking at projections across streams: https://martendb.io/documentation/events/projections/custom/ You could certainly join the streams in a more "manual fashion" (load multiple streams, join evenst & push them to a new stream) to create a new event stream from 2 or more streams... Just requires thinking a bit about event ordering and whether running on a live system or not.
